lib/deepcrawl.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+const BPromise = require('bluebird'),
+  schema = require('../schemas/v2.0.0'),
+  needle = BPromise.promisifyAll(require('needle')),
+  humps = require('humps'),
+  _ = require('lodash');
+
+class DeepCrawl {
+  constructor(cfg) {
+    cfg = cfg || {};
+
+    this.apiId = cfg.apiId;
+    this.apiKey = cfg.apiKey;
+    this.getSessionToken = cfg.getSessionToken;
+    this.accountId = cfg.accountId;
+
+    this.schema = cfg.schema || schema;
+    this.version = this.schema.version;
+    this.baseUrl = this.schema.baseUrl;
+  }
+
+  createEndpoints (schema) {
+    const API = {};
+    for (const resource in schema.resources) {
+      API[resource] = {};
+    }
+    return API;
+  }
```

Comment:
I would nuke this method, I don't think it's needed. See comments below in `addMethodsToEndpoints`